libbeat/beat/beat.go

```diff
@@ -193,5 +211,22 @@ func (b *Beat) Run() {
 
 // Stop calls the beater Stop action.
 func (beat *Beat) Stop() {
+	logp.Info("Stopping Beat")
 	beat.BT.Stop()
+	close(beat.done)
+}
+
+// Reload the beat for example with a fresh configuration
+// This requires that stop and cleanup properly shuts down the beat
```

Comment:
Do we want to rely on the fact that all Beats are able to shut down gracefully? I'd be afraid to rely on that, it's just hard to test and easy to break. Would be good to have some sort of automatic watchdog that detects somehow if the reload has failed kills the whole process. 

In all deployment situations (except from running it directly in the command line) we have some sort of supervisor (either god or systemd, upstart, etc.).